cmd/provenanced/cmd/clientconfig.go

```diff
@@ -0,0 +1,100 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/provenance-io/provenance/cmd/provenanced/config"
+	tmcli "github.com/tendermint/tendermint/libs/cli"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+)
+
+// Cmd returns a CLI command to interactively create an application CLI
+// config file.
+func ClientConfigCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "config <key> [value]",
+		Short: "Create or query an application CLI configuration file",
+		RunE:  runClientConfigCmd,
+		Args:  cobra.RangeArgs(0, 2),
+	}
+	return cmd
+}
+
+func runClientConfigCmd(cmd *cobra.Command, args []string) error {
+	clientCtx := client.GetClientContextFromCmd(cmd)
+	configPath := filepath.Join(clientCtx.HomeDir, "config")
+
+	conf, err := config.GetClientConfig(configPath, clientCtx.Viper)
+	if err != nil {
+		return fmt.Errorf("couldn't get client config: %v", err)
+	}
+
+	switch len(args) {
+	case 0:
+		// print all client config fields to stdout
+		s, err := json.MarshalIndent(conf, "", "\t")
+		if err != nil {
+			return err
+		}
+		cmd.Println(string(s))
+
+	case 1:
+		// it's a get
+		key := args[0]
+
+		switch key {
+		case flags.FlagChainID:
+			cmd.Println(conf.ChainID)
+		case flags.FlagKeyringBackend:
+			cmd.Println(conf.KeyringBackend)
+		case tmcli.OutputFlag:
+			cmd.Println(conf.Output)
+		case flags.FlagNode:
+			cmd.Println(conf.Node)
+		case flags.FlagBroadcastMode:
+			cmd.Println(conf.BroadcastMode)
+		default:
+			err := errUnknownConfigKey(key)
+			return fmt.Errorf("couldn't get the value for the key: %v, error:  %v", key, err)
+		}
+
+	case 2:
+		// it's set
+		key, value := args[0], args[1]
+
+		switch key {
+		case flags.FlagChainID:
+			conf.SetChainID(value)
+		case flags.FlagKeyringBackend:
+			conf.SetKeyringBackend(value)
+		case tmcli.OutputFlag:
+			conf.SetOutput(value)
+		case flags.FlagNode:
+			conf.SetNode(value)
+		case flags.FlagBroadcastMode:
+			conf.SetBroadcastMode(value)
+		default:
+			return errUnknownConfigKey(key)
+		}
+
+		confFile := filepath.Join(configPath, "client.toml")
+		if err := config.WriteConfigToFile(confFile, conf); err != nil {
```

Comment:
Nit: It'd be nice if `"client.toml"` were only defined in one place. It's also defined in `config/client.go#ReadFromClientConfig`. I'm not really sure where best to set it, though, so just kind of talking I guess.

Reply:
The config/client.go is copied directly from the SDK and not modified other than the keyring backend default that had to be changed to allow our chain development networks to function.  I would rather not refactor it too much.  We could restructure it more in #403 if needed though.